overlord/hookstate/ctlcmd/get.go

```diff
@@ -291,15 +291,38 @@ func (c *getCommand) getInterfaceSetting(context *hookstate.Context, plugOrSlot
 	st.Lock()
 	defer st.Unlock()
 
+	var slotRef interfaces.SlotRef
+	var plugRef interfaces.PlugRef
+	if err = attrsTask.Get("slot", &slotRef); err != nil {
+		return fmt.Errorf(i18n.G("internal error: cannot get slot reference from appropriate task"))
+	}
+	if err = attrsTask.Get("plug", &plugRef); err != nil {
+		return fmt.Errorf(i18n.G("internal error: cannot get plug reference from appropriate task"))
+	}
+
 	attributes := make(map[string]interface{})
 	if err = attrsTask.Get(which, &attributes); err != nil {
 		return fmt.Errorf(i18n.G("internal error: cannot get %s from appropriate task"), which)
 	}
 
 	return c.printValues(func(key string) (interface{}, bool, error) {
-		if value, ok := attributes[key]; ok {
+		subkeys, err := config.ParseKey(key)
+		if err != nil {
+			return nil, false, err
+		}
+
+		var value interface{}
+		err = config.GetFromChange(context.SnapName(), subkeys, 0, attributes["static"].(map[string]interface{}), &value)
```

Comment:
Instead of deep-nesting, we can have "plug-dynamic" and "plug-static" in place of "plug-attrs".

Also, the logic above seems dangerous. It's type-asserting attributes["static"].. if it's missing for whatever reason, this logic will panic due to the type assertion.